lib/jsdom/living/nodes/HTMLInputElement-impl.js

```diff
@@ -546,24 +546,24 @@ class HTMLInputElementImpl extends HTMLElementImpl {
 
     if (!this._isStepAligned(value)) {
       value = this._stepAlign(value, /* roundUp = */ isUp);
-    }
-
-    let delta = n * allowedValueStep;
-    if (!isUp) {
-      delta *= -1;
-    }
-    value += delta;
+    } else {
+      let delta = n * allowedValueStep;
+      if (!isUp) {
+        delta *= -1;
+      }
+      value += delta;
 
-    if (min !== null && value < min) {
-      value = this._stepAlign(min, /* roundUp = */ true);
-    }
+      if (min !== null && value < min) {
```

Comment:
The [spec](https://html.spec.whatwg.org/multipage/input.html#dom-input-stepup) says that while the block of code above should be in the `else` branch, this and the following `if`-statements should not be.

Reply:
Good point -- stepAlign doesn't include over-run checks itself. I've updated that, and will look at adding a test.